graphql/resolvers.go

```diff
@@ -2530,6 +2524,129 @@ func (r *queryResolver) BbGetCreatedTickets(ctx context.Context, taskID string)
+	}
+
+	activatedVersions, err := model.GetMainlineCommitVersionsWithOptions(projectId, opts)
+	if err != nil {
+		return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("Error getting activated versions %s", err.Error()))
+	}
+
+	opts = model.MainlineCommitVersionOptions{
+		Activated:       false,
+		SkipOrderNumber: utility.FromIntPtr(options.SkipOrderNumber),
+	}
+	unactivatedVersions, err := model.GetMainlineCommitVersionsWithOptions(projectId, opts)
+	if err != nil {
+		return nil, ResourceNotFound.Send(ctx, fmt.Sprintf("Error getting unactivated versions %s", err.Error()))
+	}
+
+	versions := append(activatedVersions, unactivatedVersions...)
+	sort.Slice(versions, func(i, j int) bool {
+		return versions[i].RevisionOrderNumber > versions[j].RevisionOrderNumber
+	})
+	var mainlineCommits MainlineCommits
+	activatedVersionCount := 0
+	for _, v := range versions {
+		if activatedVersionCount == *options.Limit {
+			break
+		}
+		apiVersion := restModel.APIVersion{}
+		err = apiVersion.BuildFromService(&v)
+		if err != nil {
+			return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error building APIVersion from service: %s", err.Error()))
+		}
+
+		// If we try to access a version that does not have the Activated flag we must manually verify it
+		// After we verify if a version is activated we cache that field so subsequent queries are faster.
+		if v.Activated == nil {
+			defaultSort := []task.TasksSortOrder{
+				{Key: task.DisplayNameKey, Order: 1},
+			}
+			opts := data.TaskFilterOptions{
+				Statuses:        []string{},
+				BaseStatuses:    []string{},
+				Variants:        []string{},
+				TaskNames:       []string{},
+				Page:            0,
+				Limit:           0,
+				FieldsToProject: []string{},
+				Sorts:           defaultSort,
+			}
+			tasks, _, err := r.sc.FindTasksByVersion(v.Id, opts)
+			if err != nil {
+				return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error fetching tasks for version %s : %s", v.Id, err.Error()))
+			}
+			if !task.AnyActiveTasks(tasks) {
+				err = v.SetNotActivated()
+				if err != nil {
+					return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error updating version activated status for %s, %s", v.Id, err.Error()))
+				}
+			} else {
+				err = v.SetActivated()
+				if err != nil {
+					return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error updating version activated status for %s, %s", v.Id, err.Error()))
+				}
+			}
+		}
+		mainlineCommitVersion := MainlineCommitVersion{}
+
+		if utility.FromBoolPtr(v.Activated) {
+			activatedVersionCount++
+			mainlineCommits.NextPageOrderNumber = &v.RevisionOrderNumber
+			mainlineCommitVersion.Version = &apiVersion
+			mainlineCommits.Versions = append(mainlineCommits.Versions, &mainlineCommitVersion)
+		} else {
+			if len(mainlineCommits.Versions) > 0 {
+				lastMainlineCommit := mainlineCommits.Versions[len(mainlineCommits.Versions)-1]
+				if lastMainlineCommit.RolledUpVersions != nil {
```

Comment:
You can check this in the go playground, but I believe appending to a nil list does work. Otherwise if there aren't any rolled up versions yet, you're just omitting the version completely at this step.

Reply:
Thanks! I added some comments and moved the logic a little bit to make it more clear what this is doing. I also updated the graphql tests to capture this scenario.

Reply:
thanks for these comments!